fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/logical/MultiJoin.java

```diff
@@ -185,7 +219,11 @@ public Void visitLogicalJoin(LogicalJoin<Plan, Plan> join, Void context) {
         join.left().accept(this, context);
         join.right().accept(this, context);
 
-        join.getCondition().ifPresent(cond -> conjuncts.addAll(ExpressionUtils.extractConjunction(cond)));
+        conjunctsForAllHashJoins.addAll(join.getHashJoinPredicates());
+        if (join.getOtherJoinCondition().isPresent()) {
+            conjunctsForAllHashJoins.addAll(ExpressionUtils.extractConjunction(join.getOtherJoinCondition().get()));
```

Comment:
this name is not very good since it could have other conditions

Reply:
before apply FindHashConditionForJoin rule, all the conditions are put into otherJoinCondition. FindHashConditionForJoin rule will extract hash join predicates from otherJoinCondition.
MultiJoin applied before FindHashConditionForJoin rule.